src/cpu/x64/asm.rs

```diff
@@ -1128,6 +1136,30 @@ fn sse_float_reg_freg(buf: &mut MacroAssembler, dbl: bool, op: u8, x64: u8, dest
     emit_modrm(buf, 0b11, dest.and7(), src.and7());
 }
 
+fn sse_movd_reg_freg(buf: &mut MacroAssembler, dbl: bool, op: u8, dest: FReg, x64: u8, src: Reg) {
+    emit_op(buf, 0x66);
+
+    if x64 != 0 || dest.msb() != 0 || src.msb() != 0 {
+        emit_rex(buf, x64, dest.msb(), 0, src.msb());
+    }
+
+    emit_op(buf, 0x0f);
+    emit_op(buf, op);
+    emit_modrm(buf, 0b11, dest.and7(), src.and7());
+}
+
+fn sse_movd_freg_reg(buf: &mut MacroAssembler, dbl: bool, op: u8, x64: u8, dest: Reg, src: FReg) {
```

Comment:
dbl is not needed as argument right? Could you also please make x64 the second argument here as well.